client/adjudicate.go

```diff
@@ -243,6 +243,9 @@ func (c *Channel) ForceUpdate(ctx context.Context, updater func(*channel.State))
 // Returns ChainNotReachableError if the connection to the blockchain network
 // fails when sending a transaction to / reading from the blockchain.
 func (c *Channel) Settle(ctx context.Context, secondary bool) (err error) {
+	// Save the phase in which settle was called.
+	oldPhase := c.Phase()
```

Comment:
This unfortunately adds additional complexity to the settle function. It would be better to ensure that when creating a channel object, it is already setup correctly so that we never need to check something like `if not setup correctly, don't tear down that part`. The purpose here is to check whether the account usage was incremented or not. Could you instead increment the account usage directly after channel creation, or maybe even slightly before? (otherwise it would be problematic to garbage collect the accounts during channel setup!) an alternative would also be to remove the account increment decrement logic completely.

Reply:
Done. I now increment the account usage directly after the failed funding and thus before calling settle.